driver/pace/driver/state.py

```diff
@@ -152,7 +159,15 @@ def _overwrite_state_from_restart(
     df = xr.open_dataset(path + f"/{restart_file_prefix}_{rank}.nc")
     for _field in fields(type(state)):
         if "units" in _field.metadata.keys():
-            state.__dict__[_field.name].data[:] = df[_field.name].data[:]
+            if is_gpu_backend:
+                if "physics" in restart_file_prefix:
+                    state.__dict__[_field.name][:] = cp.asarray(df[_field.name].data[:])
```

Comment:
is there a reason we don't use gt4py_util.asarray here?

Reply:
I did try that but got this `ValueError: non-scalar numpy.ndarray cannot be used for fill`. But maybe it would work if I specify `to_type`?

Reply:
Okay yes that was obvious, I overlooked the api for asarray. Switched to `gt4py_util.asarray`.